### DISCUSSION TITLE
Templating Issues and no power analytics for the Shelly 2.5 devices.

### DISCUSSION THREAD

**Discussion by chrishome7** [arendst/Tasmota#11574] [Q&A] [answered]
I'm new and i'm not sure if this is an issue that needs reporting. I think it may.  Any (friendly) advice would be cool.

After flashing multiple UL listed Shelly 2.5's (via mgos-to-tasmota) and manually with pre compiled v9.3.1(Tasmota) there are no power analytics. (Core/SDK Version 2_7_4_9/2.2.2-dev(38a443e))

The Devices Automatically switch themselves to the new 9.1+ template listed in the repository and doesn't seem to properly support the power monitoring features of the shelly 2.5. 
{"NAME":"Shelly 2.5","GPIO":[320,0,32,0,224,193,0,0,640,192,608,225,3456,4736],"FLAG":0,"BASE":18} 

The OG template listed on Tasmota Git  (https://tasmota.github.io/docs/devices/Shelly-2.5/)
{"NAME":"Shelly 2.5","GPIO":[56,0,17,0,21,83,0,0,6,82,5,22,156],"FLAG":2,"BASE":18}
Does properly initiate the desired power monitoring fields (in the UI), however, said devices automatically revert partaily to the new template and does not recognize the sensors or populate the analytics.  After a power cycle the devices automatically revert back fully to the "new" template and the UI no longer displays the defunct power motoring fields. 
Status 8 only shows Temp sensors. Status 9 throws an error.
I've reset 5,6, etc and re-update the devices many times.  Same Results.   Regardless of the alteration (including Template and module name) the devices automagicly retard to the same faulty IO template.

**Comment by sfromis** [comment, marked as answer]
My Shelly 2.5 devices work perfectly fine with the current template directly from the repository. This includes `Status 9` providing the expected output, and energy monitoring data included in status 10 (or 8). Not sure exactly what happens, I'd try a full "factory reset" of Tasmota, `Reset 1` to revert to initial configuration, and configure what's needed, of course including template from the repository.

As for new vs old template, Tasmota will automatically convert gpio component numbers in outdated templates to current. Old numbering scheme is no longer meaningful with recent Tasmota versions.

**Reply by barbudor** [reply]
@chrishome7 
The energy meter on the shelly is connected in I2C.
Please check 

- That you have I2C SDA on GPIO12, and I2C SCL on GPIO14
- That `i2cscan` command returns a device found at address 0x38
- That `i2cdriver` shows that driver 7 is present and enabled
